terraform/modules/origin/main.tf

```diff
@@ -103,7 +103,7 @@ data "aws_route53_zone" "public" {
 }
 
 resource "aws_route53_record" "origin_alb" {
-  zone_id = data.aws_route53_zone.public.zone_id
+  zone_id = var.public_zone_id
```

Comment:
We should probably be more consistent about when we pass zone IDs vs domain names. But that's something to think about later.